server/providers/emails/mailgun/index.ts

```diff
@@ -0,0 +1,15 @@
+import nodemailer from 'nodemailer'
+import Mailer from 'nodemailer/lib/mailer'
+import mg from 'nodemailer-mailgun-transport'
+import { Provider, Mail } from '../../../email/provider'
+
+export default class MailgunProvider implements Provider {
+  private mailer: Mailer
+  constructor (options: mg.Options) {
+    this.mailer = nodemailer.createTransport(mg(options))
```

Comment:
Secrets (like the API key) need to be able to be read from environment variables.